src/modules/workspaces/workspaces.usecase.spec.ts

```diff
@@ -0,0 +1,343 @@
+    });
+
+    it('When invitation space limit exceeds the assignable space, then it should throw', async () => {
+      const invitedUser = newUser();
+      const workspace = newWorkspace();
+      const invitedUserEmail = 'newUser@example.com';
+      const spaceLeft = 1024 * 1024 * 10;
+
+      jest
+        .spyOn(workspaceRepository, 'findById')
+        .mockResolvedValueOnce(workspace);
+      jest
+        .spyOn(userUsecases, 'findByEmail')
+        .mockResolvedValueOnce(invitedUser);
+      jest
+        .spyOn(userUsecases, 'findPreCreatedByEmail')
+        .mockResolvedValueOnce(null);
+      jest.spyOn(service, 'isWorkspaceFull').mockResolvedValueOnce(false);
+      jest
+        .spyOn(service, 'getAssignableSpaceInWorkspace')
+        .mockResolvedValueOnce(BigInt(spaceLeft));
+      jest.spyOn(workspaceRepository, 'findInvite').mockResolvedValueOnce(null);
+
+      await expect(
+        service.inviteUserToWorkspace(user, workspace.id, {
+          invitedUser: invitedUserEmail,
+          spaceLimit: BigInt(spaceLeft + 1),
+          encryptionKey: 'encryptionKey',
+          encryptionAlgorithm: 'RSA',
+        }),
+      ).rejects.toThrow(BadRequestException);
+    });
+  });
+
+  describe('isWorkspaceFull', () => {
+    const workspaceId = 'workspace-id';
+    it('When workspace has slots left, then workspace is not full', async () => {
+      jest
+        .spyOn(workspaceRepository, 'getWorkspaceUsersCount')
+        .mockResolvedValue(5);
+      jest
+        .spyOn(workspaceRepository, 'getWorkspaceInvitationsCount')
+        .mockResolvedValue(4);
+
+      const isFull = await service.isWorkspaceFull(workspaceId);
+      expect(isFull).toBe(false);
+    });
+    it('When workspace has slots left, then workspace is full', async () => {
+      jest
+        .spyOn(workspaceRepository, 'getWorkspaceUsersCount')
+        .mockResolvedValue(10);
+      jest
+        .spyOn(workspaceRepository, 'getWorkspaceInvitationsCount')
+        .mockResolvedValue(0);
+
+      const isFull = await service.isWorkspaceFull(workspaceId);
+      expect(isFull).toBe(true);
+    });
+  });
+
+  describe('getAssignableSpaceInWorkspace', () => {
+    const workspace = newWorkspace();
+    const workspaceDefaultUser = newUser();
+    it('When there is space left, then it should return the correct space left', async () => {
+      jest.spyOn(networkService, 'getLimit').mockResolvedValue(1000000);
+      jest
+        .spyOn(workspaceRepository, 'getTotalSpaceLimitInWorkspaceUsers')
+        .mockResolvedValue(BigInt(500000));
+      jest
+        .spyOn(workspaceRepository, 'getSpaceLimitInInvitations')
+        .mockResolvedValue(BigInt(200000));
+
+      const assignableSpace = await service.getAssignableSpaceInWorkspace(
+        workspace,
+        workspaceDefaultUser,
+      );
+      expect(assignableSpace).toBe(BigInt(300000));
+    });
+
+    it('When there is no space left, then it should return 0', async () => {
```

Comment:
As this function does not have multiple ways, just once, which is an aritmetical operation, this test is not covering anything new, as far as we check that the operation is being done properly (the first test)